### PR TITLE
Clarify Android session management steps

### PR DESCRIPTION
**Pull request by aeromusek**
@sojanpr for technical review, and @aaustin for oversight (because this explicitly reverts some of the changes in a4a649405c84e733d6f8548e9eed546d444ba255 and I'm not sure if there was a reason for those that I'm missing)

### CODE REVIEW

File: pages/getting-started/sdk-integration-guide.md [hunk 1]
```diff
@@ -366,15 +366,18 @@ Branch opens your app by using its URI scheme (`yourapp://`), which should be un
...
-Simply create a Branch instance in your `Application#onCreate()` method:
+In your **AndroidManifest.xml**, add an `android:name` parameter to your `<application>` definition:
 
-{% highlight java %}
-public void onCreate() {
-    super.onCreate();
-    Branch.getAutoInstance(this);
-}
+{% highlight xml %}
+<application
```
Comment (aaustin) [inline pages/getting-started/sdk-integration-guide.md]: This is very uncommon now a days. Ever developer has an application class as default when creating a new project. It should not be our default recommendation.

Comment (aaustin) [inline pages/getting-started/sdk-integration-guide.md]: (that's why I changed it before)

Comment (aeromusek) [inline pages/getting-started/sdk-integration-guide.md]: In this version, it's now on the Advanced page as the final option if all others fail. Is that OK? I can push this up to Staging if you want to see it

Comment (aaustin) [inline pages/getting-started/sdk-integration-guide.md]: Sorry. Maybe I was not clear. 95%+ of Android of developers who integrate Branch will need to use the code you're moving to the advanced section. So, with this change, every developer must visit the advanced section and then skip back to the basic one to complete the guide. While more complicated, the current prioritization of integration methods before this change is the correct one.

Comment (aeromusek) [inline pages/getting-started/sdk-integration-guide.md]: Ah, I get it now. Stand by for an update...

**Comment by aeromusek** [comment]
@aaustin should be better now

File: pages/getting-started/sdk-integration-guide.md (hunk 1 above)
Comment (sojanpr) [inline pages/getting-started/sdk-integration-guide.md]: Alex there no application class created when you create a project. Some one has to create it manually and add the entry in the manifest. So I guess most of the applications won't have application class as they create the project

**Comment by aaustin** [comment]
@sojanpr Facebook and Firebase both have the custom application assumed in their integration. I think it's safe for us to do this too. Plus, it allows people to use the getTestAutoInstance as well. 👍

**Pull request merged**